client/app/bundles/course/assessment/submission/components/answers/TextResponse.jsx

```diff
@@ -58,6 +82,75 @@ function renderTextResponseSolutions(question) {
   /* eslint-enable react/no-array-index-key */
 }
 
+function renderTextResponseComprehensionPoint(point) {
+  /* eslint-disable react/no-array-index-key */
+  return (
+    <React.Fragment>
+      <br />
+      <h5><FormattedMessage {...translations.point} /></h5>
+      <Table selectable={false}>
+        <TableBody displayRowCheckbox={false}>
+          <TableRow>
+            <TableRowColumn><FormattedMessage {...translations.pointGrade} /></TableRowColumn>
+            <TableRowColumn>{point.pointGrade}</TableRowColumn>
+          </TableRow>
+          <TableRow>
+            <TableHeaderColumn><FormattedMessage {...translations.type} /></TableHeaderColumn>
+            <TableHeaderColumn><FormattedMessage {...translations.solution} /></TableHeaderColumn>
+          </TableRow>
+          {point.solutions.map((solution, index) => (
+            <TableRow key={index}>
+              <TableRowColumn><FormattedMessage {...translations[solution.solutionType]} /></TableRowColumn>
+              <TableRowColumn style={{ whiteSpace: 'pre-wrap' }}>{solution.solution}</TableRowColumn>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+    </React.Fragment>
+  );
+  /* eslint-enable react/no-array-index-key */
+}
+
+function renderTextResponseComprehensionGroup(group) {
+  /* eslint-disable react/no-array-index-key */
+  return (
+    <React.Fragment>
+      <br />
+      <h4><FormattedMessage {...translations.group} /></h4>
+      <Table selectable={false}>
+        <TableBody displayRowCheckbox={false}>
+          <TableRow>
+            <TableRowColumn><FormattedMessage {...translations.maximumGroupGrade} /></TableRowColumn>
+            <TableRowColumn>{group.maximumGroupGrade}</TableRowColumn>
+          </TableRow>
+          {group.points.map((point, index) => (
+            <TableRow key={index}>
```

Comment:
Instead of using `eslint-disable`, could you pass the `id`s of point and group in the JSON and use it for the key?

Reply:
Done. I have also changed for the original TextResponse solutions (non-compre).